modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/RebalanceRaftGroupEventsListener.java

```diff
@@ -306,6 +307,8 @@ private void doOnNewPeersConfigurationApplied(PeersAndLearners configuration) {
                     )
             ).get();
 
+            Set<Assignment> calculatedAssignments = calculateAssignmentsFn.apply(tablePartitionId).get();
```

Comment:
So, just blocking get(). Nice way to get a deadlock.

Reply:
> We have already make get on ms.getAll in a few lines above
Yep, and we also have `// TODO: https://issues.apache.org/jira/browse/IGNITE-17592` Remove synchronous wait in a few more lines above.
Within which thread are we going to wait? Is it a raft one?

Reply:
No, this is a thread from `RebalanceRaftGroupEventsListener#rebalanceScheduler`